automated install/basic-install.sh

```diff
@@ -1357,27 +1424,61 @@ check_service_active() {
+    if check_service_active "systemd-resolved"; then
+      # Check if DNSStubListener is enabled
+      echo -e "  ${INFO} Testing if systemd-resolved DNSStub-Listener is active"
+      if ( grep -E '#?DNSStubListener=yes' /etc/systemd/resolved.conf &> /dev/null ); then
+        # Disable the DNSStubListener to unbind it from port 53
+        # Note that this breaks dns functionality on host until dnsmasq/ftl are up and running
+        echo -e "  ${TICK} Disabling systemd-resolved DNSStubListener"
+        # Make a backup of the original /etc/systemd/resolved.conf
+        # (This will need to be restored on uninstallation)
+        ${SUDO} sed -r -i.orig 's/#?DNSStubListener=yes/DNSStubListener=no/g' /etc/systemd/resolved.conf
+        echo -e "${TICK}  Restarting systemd-resolved DNSStubListener"
+        ${SUDO} systemctl reload-or-restart systemd-resolved
+      else
+        echo -e "  ${INFO} Systemd-resolved does not need to be restarted"
+      fi
+    fi
+  else
+    echo -e "  ${INFO} Systemd-resolved is not enabled"
+  fi
+}
+
+disable_dnsmasq () {
+  # Check if dnsmasq flag is present.
+  if [[ $dnsmasq_flag = "true" ]]; then
+    if ( grep -E '#?dns=dnsmasq' /etc/NetworkManager/NetworkManager.conf &> /dev/null ); then
+      # Disable dnsmasq from restarting when Network Manager restarts
+      # Note that this breaks dns functionality on host until FTLDNS is up and running
+      echo -e "  ${TICK} Disabling dnsmasq from restarting when Network manager starts/restarts"
+      # Make a backup of the original /etc/NetworkManager/NetworkManager.conf
       # (This will need to be restored on uninstallation)
-      sed -r -i.orig 's/#?DNSStubListener=yes/DNSStubListener=no/g' /etc/systemd/resolved.conf
-      echo -e " and restarting systemd-resolved"
-      systemctl reload-or-restart systemd-resolved
+      ${SUDO} sed -r -i.orig '/#?dns=dnsmasq/d' /etc/NetworkManager/NetworkManager.conf
+      echo -e "  ${TICK} Restarting Network manager"
+      ${SUDO} systemctl reload-or-restart NetworkManager
     else
-      echo -e "${OVER}  ${INFO} Systemd-resolved does not need to be restarted"
+      # Disabling dnsmasq via systemctl
+      echo -e "  ${INFO} Disabling dnsmasq via systemctl"
+      ${SUDO} systemctl disable dnsmasq &> /dev/null
+    fi
+    #setting dnsmasq_flag to false in order for the next check to validate or not
+    dnsmasq_flag=false
+    echo -e "${OVER}  ${INFO} Checking if dnsmasq is still running"
+    silent53check
+    if [[ $dnsmasq_flag = "true" ]]; then
+      echo -e "${OVER}  ${EXCL} dnsmasq still active, this is most likely due to the fact that ${COL_LIGHT_RED}dnsmasq
+      was loaded via a non convetional method. This might cause future conflicts with FTLDNS${COL_NC}"
+      ${SUDO} pkill dnsmasq
+      echo -e "${OVER}  ${TICK} dnsmasq process killed"
     fi
   else
-    echo -e "${OVER}  ${INFO} Systemd-resolved is not enabled"
+    echo -e "${INFO} dnsmasq is not enabled"
```

Comment:
If you are echoing out here, you should also echo out at the start of the function.